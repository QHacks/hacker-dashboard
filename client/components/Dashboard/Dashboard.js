import { actionCreators } from '../../HackerStore';
import { Container, Divider, Dropdown, Grid, Header, Image, List, Menu, Segment } from 'semantic-ui-react';
import React, { Component } from 'react';
import Emoji from 'react-emoji-render';
import { connect } from 'react-redux';

const { logout, getUsers } = actionCreators;

class Dashboard extends Component {

	handleUsersClick() {
		this.props.getUsers();
	}

	handleLogoutClick() {
		this.props.logout();
	}

	render() {
		return (
			<div>
				<Menu stackable size='large'>
					<Menu.Item>
						<img src={require('../../assets/img/qhacks-tricolor-logo.svg')} />
					</Menu.Item>

					<Menu.Menu position='right'>
						<Menu.Item>
							Logout
						</Menu.Item>
					</Menu.Menu>
				</Menu>

				<Container text style={{ marginTop: '3em' }}>
					<Segment raised>
						<Header as='h2'><Emoji text="Congratulations, your application is in! :tada: :heart:" /></Header>
						<p>Your application to attend QHacks 2018 has been submitted! So, what are the next steps? </p>
						<p>We will be accepting applications until January 8th 2018, later on that week we will be sending emails to all of the applicants letting them know their application status.</p>
						<p>If you are accepted, the next step will be to have you RSVP right here in the dashboard, you must complete this step within one week of receiving your acceptance letter. The RSVP confirms that you will be attending and provides us with more information about you to make your experience the best it can be. Some of the information we'll ask you for are things like t-shirt size, dietary restrictions, workshops you are interested in attending, actvities you might be interested in and much more.</p>
						<p>What if I don't get accepted? We'll have a waiting list that you can enter into from the dashboard and it will be a first come first serve basis. We really wish that we could accept everyone but we receive thousands of applications and just do not have the infrastructure to maintain the quality and safety of our event. Every year there are always those applicants that reach out to us asking if there is anything we can do to help them and our advice is to just make sure you are looking for that email and jump on the waiting list as soon as you can, we can't make any exceptions because it wouldn't be fair.</p>
						<p>While you wait to hear from us there are some things you can do, such as:</p>
						<ul>
							<li>Like us on <a href="">Facebook</a></li>
							<li>Follow us on <a href="">Twitter</a></li>
							<li>Follow us on <a href="">Instagram</a></li>
						</ul>
						<p>We will be having contests for free tech on our social media so do not miss out! <Emoji text=":watch: :iphone:" /></p>
						<p>Finally, we want to say thank you for your excitement about QHacks! With the participation of people like you, we can all continue towards our goal of making Queenâ€™s and Canada recognized for its talent and enthusiasm for innovation and creation.</p>
					</Segment>
				</Container>
			</div>
		);
	}
}

export default connect((state) => state, { logout, getUsers })(Dashboard);
