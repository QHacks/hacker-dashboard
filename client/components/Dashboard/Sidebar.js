import { Menu, Sidebar } from 'semantic-ui-react';
import React from 'react';

export default ({ children, ...rest }) => (
    <Sidebar as={Menu}
             animation="overlay"
             width="thin"
             icon="labeled"
             vertical
             inverted
             {...rest}>
        {children}
    </Sidebar>
);
