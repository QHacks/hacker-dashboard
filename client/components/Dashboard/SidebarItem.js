import React, { Component } from 'react';
import { connect } from 'react-redux';
import { actionCreators } from '../../HackerStore/HackerActions';
import { Link } from 'react-router-dom';
import { Menu } from 'semantic-ui-react';

const { toggleSidebarVisibility } = actionCreators;

class SidebarItem extends Component {
    render() {
        const { children, toggleSidebarVisibility, ...rest } = this.props;
        return (
            <Menu.Item as={Link}
                       onClick={() => toggleSidebarVisibility()}
                       {...rest}>
                {children}
            </Menu.Item>
        );
    }
}

function mapStateToProps(state, ownProps) {
    return { ...ownProps };
}

export default connect(mapStateToProps, { toggleSidebarVisibility })(SidebarItem);
