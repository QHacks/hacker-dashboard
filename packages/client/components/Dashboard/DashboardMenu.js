import React, { Component } from "react";
import ActionButton from "../ActionButton/ActionButton";
import ContentWrapper from "../ContentWrapper/ContentWrapper";
import crown from "../../assets/img/qhacks-tricolor-logo.svg";
import wordmark from "../../assets/img/qhacks-wordmark-colored.svg";

class DashboardMenu extends Component {
  render() {
    return (
      <header
        css={`
          width: 100%;
          height: 84px;
          line-height: 44px;
          padding: 20px 0;
          position: absolute;
          top: 0;
          left: 0;
          border-bottom: 1px solid #a4b3bf;
        `}
      >
        <ContentWrapper>
          <div
            css={`
              display: inline-block;
            `}
          >
            <img
              src={crown}
              css={`
                max-height: 32px;
                margin-right: 6px;
              `}
              alt="QHacks crown"
            />
            <img
              src={wordmark}
              css={`
                max-height: 24px;
              `}
              alt="QHacks wordmark"
            />
          </div>
          <div
            css={`
              display: inline-block;
              float: right;
              line-height: 44px;
            `}
          >
            <ActionButton
              style={`font-size: 16px;`}
              type="rounded"
              internal
              link="/logout"
            >
              Log out
            </ActionButton>
          </div>
        </ContentWrapper>
      </header>
    );
  }
}

export default DashboardMenu;
