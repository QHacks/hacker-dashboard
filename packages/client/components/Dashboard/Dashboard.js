import React, { Component } from "react";

import ContentWrapper from "../ContentWrapper/ContentWrapper";
import StatusReport from "../StatusReport/StatusReport";
import circuits from "../../assets/img/circuits.png";
import Events from "./DashboardSection/Events";
import DashboardMenu from "./DashboardMenu";

class Dashboard extends Component {
  renderStatusReports() {
    if (this.props.location.state) {
      const { alert } = this.props.location.state;

      if (alert) {
        return <StatusReport {...alert} />;
      }
    }
  }

  render() {
    return (
      <div
        css={`
          margin-top: 84px;
          background: url(${circuits}) no-repeat center bottom;
          background-size: cover;
          padding: 40px 0;
          height: calc(100vh - 84px);
        `}
      >
        <DashboardMenu />
        <ContentWrapper>
          {this.renderStatusReports()}
          <Events />
        </ContentWrapper>
      </div>
    );
  }
}

export default Dashboard;
