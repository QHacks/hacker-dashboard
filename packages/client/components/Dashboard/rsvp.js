import React, { Component } from "react";
import DashboardMenu from "./DashboardMenu";
import { Link } from "react-router-dom";
import { mobileMaxWidth } from "../../assets/constants";
import Checkbox from "react-simple-checkbox";
import { checkboxOptions } from "../../assets/constants";
import ActionButton from "../ActionButton/ActionButton";
import Select from "react-select";
import arrow from "../../assets/img/arrow.svg";

class Rsvp extends Component {
  constructor(props) {
    super(props);
    this.state = {
      shirtSize: this.toOptions("M"),
      pronoun: "",
      dietaryRestrictions: "",
      areasOfInterest: [],
      travelOrigin: "",
      qhacksBus: this.toOptions("no")
    };
  }
  toOptions(str) {
    return {
      label: str,
      value: str
    };
  }
  setSelectAnswer(field) {
    return (answer) =>
      this.setState((state) => {
        state[field] = answer;
        return state;
      });
  }

  sendRsvp() {
    // pass
  }

  setInterest(interest, bool) {
    this.setState((state) => ({
      areasOfInterest: bool
        ? [...state.areasOfInterest, interest]
        : state.areasOfInterest.filter((x) => x !== interest)
    }));
  }

  render() {
    const shirtSizeOptions = ["XS", "S", "M", "L", "XL"].map((x) =>
      this.toOptions(x)
    );
    const qhacksBusOptions = ["yes", "no"].map((x) => this.toOptions(x));

    const inputRowStyle = `
      width: 100%;
      display: flex;
      flex-direction: row;
      @media (max-width: ${mobileMaxWidth}) {
        flex-direction: column;
      }
      margin-top:12px;
      align-items: stretch;
      label, input {
          display: block;
      }
      >div{
          flex-grow: 1;
          @media (max-width: ${mobileMaxWidth}) {
            min-width: 100%;
          }
          min-width: 50%;
          padding: 4px;
      }
    `;

    return (
      <div
        css={`
          color: #252525;
          div h1,
          div h2,
          div h3 {
            color: #252525 !important;
          }
          h1 {
            font-size: 28px;
          }
          h2 {
            font-size: 24px;
          }
          p {
            margin: 12px 0;
          }
        `}
      >
        <DashboardMenu />
        <div
          css={`
            margin-top: 84px;
            padding: 28px 48px;
            background-color: white;
            box-shadow: 0 2px 16px 0 rgba(212, 217, 225, 0.4);
            a {
              display: flex;
              align-items: center;
            }
          `}
        >
          <Link to="/profile">
            <img
              src={arrow}
              css={`
                transform: rotate(180deg);
                margin-right: 8px;
              `}
            />{" "}
            Back to my dashboard
          </Link>
          <h1>RSVP to QHacks 2019</h1>
        </div>
        <div
          css={`
            padding: 56px 0;
            text-align: center;
            margin: 0 auto;
            max-width: 784px;
          `}
        >
          <h2
            css={`
              margin-bottom: 8px;
            `}
          >
            We are so excited you are coming to QHacks 2019!
          </h2>
          <p
            css={`
              font-size: 15px;
            `}
          >
            Please fill out this form so we can prepare for your stay.
          </p>
          <div
            css={`
              margin-top: 70px;
            `}
          >
            <section css={inputRowStyle}>
              <div>
                <label htmlFor="pronoun">Pronoun (optional)</label>
                <input
                  id="pronoun"
                  onChange={(e) => this.setState({ pronoun: e.target.value })}
                  value={this.state.pronoun}
                  type="text"
                />
              </div>
              <div>
                <label htmlFor="shirtSize">Shirt size</label>
                <Select
                  options={shirtSizeOptions}
                  id="shirtSize"
                  key="shirtSize"
                  className="select"
                  value={this.state.shirtSize}
                  onChange={this.setSelectAnswer("shirtSize")}
                />
              </div>
            </section>
            <section css={inputRowStyle}>
              <div>
                <label htmlFor="dietaryRestrictions">
                  Dietary restrictions
                </label>
                <input
                  id="dietaryRestrictions"
                  type="text"
                  onChange={(e) =>
                    this.setState({ dietaryRestrictions: e.target.value })
                  }
                  value={this.state.dietaryRestrictions}
                />
              </div>
            </section>
            <section
              css={`
                margin-top: 32px;
              `}
            >
              <div>
                <label>Areas of interest</label>
                <div
                  css={`
                    display: flex;
                    width: 100%;
                    padding: 16px 24px 32px;
                    > div {
                      width: 50%;
                      height: 100%;
                      display: flex;
                      flex-direction: column;
                      text-align: left;
                      label {
                        color: #252525 !important;
                        margin-left: 4px;
                        font-weight: regular;
                      }
                    }
                  `}
                >
                  <div>
                    <div>
                      <Checkbox
                        id="interest-aiMl"
                        {...checkboxOptions}
                        checked={this.state.areasOfInterest.includes("aiMl")}
                        onChange={(bool) => this.setInterest("aiMl", bool)}
                      />
                      <label htmlFor="interest-aiMl">AI/ML</label>
                    </div>
                    <div>
                      <Checkbox
                        id="interest-blockchain"
                        {...checkboxOptions}
                        checked={this.state.areasOfInterest.includes(
                          "blockchain"
                        )}
                        onChange={(bool) =>
                          this.setInterest("blockchain", bool)
                        }
                      />
                      <label htmlFor="interest-blockchain">Blockchain</label>
                    </div>
                    <div>
                      <Checkbox
                        id="interest-arVr"
                        {...checkboxOptions}
                        checked={this.state.areasOfInterest.includes("arVr")}
                        onChange={(bool) => this.setInterest("arVr", bool)}
                      />
                      <label htmlFor="interest-arVr">AR/VR</label>
                    </div>
                    <div>
                      <Checkbox
                        id="interest-mobile"
                        {...checkboxOptions}
                        checked={this.state.areasOfInterest.includes("mobile")}
                        onChange={(bool) => this.setInterest("mobile", bool)}
                      />
                      <label htmlFor="interest-mobile">Mobile</label>
                    </div>
                    <div>
                      <Checkbox
                        id="interest-webDevelopment"
                        {...checkboxOptions}
                        checked={this.state.areasOfInterest.includes(
                          "webDevelopment"
                        )}
                        onChange={(bool) =>
                          this.setInterest("webDevelopment", bool)
                        }
                      />
                      <label htmlFor="interest-webDevelopment">
                        Web Development
                      </label>
                    </div>
                  </div>
                  <div>
                    <div>
                      <Checkbox
                        id="interest-productManagement"
                        {...checkboxOptions}
                        checked={this.state.areasOfInterest.includes(
                          "productManagement"
                        )}
                        onChange={(bool) =>
                          this.setInterest("productManagement", bool)
                        }
                      />
                      <label htmlFor="interest-productManagement">
                        Product Management
                      </label>
                    </div>
                    <div>
                      <Checkbox
                        id="interest-entrepreneurship"
                        {...checkboxOptions}
                        checked={this.state.areasOfInterest.includes(
                          "entrepreneurship"
                        )}
                        onChange={(bool) =>
                          this.setInterest("entrepreneurship", bool)
                        }
                      />
                      <label htmlFor="interest-entrepreneurship">
                        Entrepreneurship
                      </label>
                    </div>
                    <div>
                      <Checkbox
                        id="interest-openSource"
                        {...checkboxOptions}
                        checked={this.state.areasOfInterest.includes(
                          "openSource"
                        )}
                        onChange={(bool) =>
                          this.setInterest("openSource", bool)
                        }
                      />
                      <label htmlFor="interest-openSource">Open Source</label>
                    </div>
                    <div>
                      <Checkbox
                        id="interest-interviewPrep"
                        {...checkboxOptions}
                        checked={this.state.areasOfInterest.includes(
                          "interviewPrep"
                        )}
                        onChange={(bool) =>
                          this.setInterest("interviewPrep", bool)
                        }
                      />
                      <label htmlFor="interest-interviewPrep">
                        Interview Prep
                      </label>
                    </div>
                    <div>
                      <Checkbox
                        id="interest-diversityInclusion"
                        {...checkboxOptions}
                        checked={this.state.areasOfInterest.includes(
                          "diversityInclusion"
                        )}
                        onChange={(bool) =>
                          this.setInterest("diversityInclusion", bool)
                        }
                      />
                      <label htmlFor="interest-diversityInclusion">
                        Diversity & Inclusion
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </section>
            <section css={inputRowStyle}>
              <div>
                <label htmlFor="travelOrigin">
                  Where are you travelling from?
                </label>
                <input
                  type="text"
                  id="travelOrigin"
                  placeholder="Location"
                  onChange={(e) =>
                    this.setState({ travelOrigin: e.target.value })
                  }
                  value={this.state.travelOrigin}
                />
              </div>
            </section>
            <section css={inputRowStyle}>
              <div>
                <label htmlFor="qhacksBus">
                  Will you be travelling using a QHacks bus?
                </label>
                <Select
                  options={qhacksBusOptions}
                  id="qhacksBus"
                  key="qhacksBus"
                  className="select"
                  value={this.state.qhacksBus}
                  onChange={this.setSelectAnswer("qhacksBus")}
                />
              </div>
            </section>
          </div>
          <p
            css={`
              color: #5b5b5b;
              font-size: 16px;
              padding-bottom: 56px;
            `}
          >
            Busses will be travelling to Kingston, ON, from the following
            locations: Toronto, Waterloo, and London
          </p>
          <ActionButton color="blue">RSVP</ActionButton>
          <br />
          <button
            css={`
              cursor: pointer;
              margin-top: 16px;
              background-color: white;
              border: none;
              color: #646464;
            `}
          >
            Withdraw Application
          </button>
        </div>
      </div>
    );
  }
}

export default Rsvp;
