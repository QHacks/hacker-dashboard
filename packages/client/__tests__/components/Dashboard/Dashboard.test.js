import React from "react";
import Dashboard from "../../../components/Dashboard/Dashboard";
import { shallow } from "enzyme";

describe("Dashboard", () => {
  it("renders correctly", () => {
    expect(shallow(<Dashboard location={{}} />)).toMatchSnapshot();
  });
});
