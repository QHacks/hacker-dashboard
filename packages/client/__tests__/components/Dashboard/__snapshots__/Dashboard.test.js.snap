// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders correctly 1`] = `
<div
  className="css-1k6iq9z"
>
  <Apollo(DashboardMenu) />
  <_default>
    <div
      className="css-1xn3edc"
    />
    <Apollo(Events) />
  </_default>
</div>
`;
